lib/msf/base/sessions/powershell.rb

```diff
@@ -65,8 +65,8 @@ def shell_command(cmd, timeout = 1800)
       if buff.match(/#{endm}/)
         # if you see the end marker, read the buffer from the start marker to the end and then display back to screen
         buff = buff.split(/#{strm}\r\n/)[-1]
-        buff.gsub!(/\nPS .*>/, '')
-        buff.gsub!(/#{endm}/, '')
+        buff = buff.split(/#{endm}/)[0]
```

Comment:
Since `endm` isn't intended to be a regex this can just be `buff.split(endm)[0]`